### PR TITLE
Caching images in the skaha redis 

### PR DESCRIPTION
**Pull request by anujachaitanya**
- Added a shellScript to fetch images from registries and store in redis
- Created CronJob to fetch the images every hour 
- Updated GetAction in order to fetch images from Redis instead of Harbor
- Updated tests accordingly
- Added redis as a dependancy to the SKAHA helm chart

CC : @abhishekghoshhh @rob-tw

### CODE REVIEW

File: deployment/helm/skaha/templates/image-caching-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  namespace: {{ .Values.skaha.namespace }}
+  name: image-caching-cronjob
+spec:
+  schedule: "0 */1 * * *"
```
Comment (at88mph) [inline deployment/helm/skaha/templates/image-caching-cronjob.yaml]: Please make this configurable in the `values.yaml` file as a new stanza:
```yaml
deployment:
  skaha:
    imageCache:
      refreshSchedule: <cron string>
```

File: deployment/helm/skaha/values.yaml [hunk 2]
```diff
@@ -170,3 +170,12 @@ storage:
       # Example is the persistentVolumeClaim below.
       # persistentVolumeClaim:
       #   claimName: skaha-pvc
+
+# For the token caching
```
Comment (at88mph) [inline deployment/helm/skaha/values.yaml]: This should read `For caching images from the Image Repository

File: deployment/helm/skaha/image-cache/cache-images.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            repo_data=$(curl -k -s "$PROJECT_URL/$project_name/repositories/$name/artifacts?detail=true&with_label=true&page_size=-1")
+
+            echo $repo_data | jq -c '.[]' | while read -r artifact; do
+                tag=$(echo $artifact | jq -r .tags[0].name)
+
+                if [ -z "$tag" ] || [ "$tag" == "null" ]; then
+                    continue
+                fi
+
+                image_id="$harborHost/$name:$tag"
```
Comment (at88mph) [inline deployment/helm/skaha/image-cache/cache-images.sh]: We display the Project Name with our images, so this should read:
`$harborHost/$project_name/$name:$tag`

**Review by at88mph — CHANGES_REQUESTED** [review]

File: deployment/helm/skaha/templates/image-caching-cronjob.yaml (hunk 1 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/templates/image-caching-cronjob.yaml]: Updated the values.yaml

File: deployment/helm/skaha/image-cache/cache-images.sh (hunk 3 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/image-cache/cache-images.sh]: Done

File: deployment/helm/skaha/image-cache/cache-images.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                # check if labels are empty
+                if [ -z "$labelArray" ] || [ "$labelArray" == "null" ]; then
+                    echo "No labels found for $image_id"
+                    continue
+                fi
+
+                labels=$(echo $artifact | jq -c [.labels[].name])
+
+                refined_artifact=$(echo $artifact | jq -c --argjson labels "$labels" --arg id "$image_id" '{id: $id, types: $labels, digest: .digest}')
+                echo $refined_artifact | redis-cli -h $REDIS_HOST -p $REDIS_PORT -x lpush "$temp_key"
```
Comment (at88mph) [inline deployment/helm/skaha/image-cache/cache-images.sh]: Sorry, one more thing.  Please change this to an `rpush` to preserve the list order from Harbor.

**Review by at88mph — CHANGES_REQUESTED** [review]

File: deployment/helm/skaha/image-cache/cache-images.sh (hunk 4 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/image-cache/cache-images.sh]: Done

**Review by at88mph — APPROVED** [review]

**Review by brianmajor — APPROVED** [review]

**Pull request merged**